runtime/crab/src/migration.rs

```diff
@@ -35,10 +35,19 @@ impl frame_support::traits::OnRuntimeUpgrade for CustomOnRuntimeUpgrade {
 
 	#[cfg(feature = "try-runtime")]
 	fn post_upgrade(_state: Vec<u8>) -> Result<(), &'static str> {
-		assert_eq!(<frame_system::BlockHash<Runtime>>::iter().count() as BlockNumber, NEW_BLOCK_HASH_COUNT);
+		assert_eq!(
+			<frame_system::BlockHash<Runtime>>::iter().count() as BlockNumber,
+			// Never purge genesis hash.
+			// So, plus 1 here.
+			NEW_BLOCK_HASH_COUNT + 1
+		);
 		assert_eq!(
 			migration::storage_iter::<()>(b"Ethereum", b"BlockHash").count() as BlockNumber,
-			NEW_BLOCK_HASH_COUNT
+			// Never purge genesis hash.
+			// So, plus 1 here.
+			//
+			// Ethereum can include current block hash, so plus 1 more here.
```

Comment:
Interesting, why the number of hashes in the Ethereum is 1 more than the substrate side?